src/Semantic/Task.hs

```diff
@@ -88,14 +64,12 @@ import qualified Semantic.Task.Files as Files
 import           Semantic.Telemetry
 import           Semantic.Timeout
 import           Serializing.Format hiding (Options)
-import           Source.Loc
-import           Source.Source (Source)
 
 -- | A high-level task producing some result, e.g. parsing, diffing, rendering. 'Task's can also specify explicit concurrency via 'distribute', 'distributeFor', and 'distributeFoldMap'
-type TaskEff
-  = TaskC
-  ( ResolutionC
+type TaskC
+  = ResolutionC
   ( Files.FilesC
+  ( ReaderC Config
```

Comment:
Passing this around separately to enable `serialize` not to depend on the entire `TaskSession`.